e2e/specs/st_image.spec.js

```diff
@@ -103,4 +105,24 @@ describe("st.image", () => {
       "I am prefixed with some meta tags"
     );
   });
+
+  it("displays a GIF image", () => {
+    cy.getIndexed(".element-container [data-testid='stImage'] img", 11)
+      .should("have.css", "height", "100px")
+      .should("have.css", "width", "100px")
+      .should("have.attr", "src")
+      .should("match", /^.*\.gif$/);
+  });
+
+  it("displays a GIF image and a caption together", () => {
+    cy.get(".element-container [data-testid='stImage']")
+      .eq(15)
+      .matchImageSnapshot("gif-with-caption");
```

Comment:
Does the gif always show the same content, or is there a possibility for flakiness here?

Reply:
I've made it so that the gif always shows the same content, no randomness involved: 
![gif](https://user-images.githubusercontent.com/20672874/193065212-5af99578-e6ef-4dea-a05d-5d6e37a048b8.gif)

Reply:
But it is still moving and the snapshot is only taking the state based on some time which might be different. So it could take the snapshot at different stages of the gif, or not?

Reply:
Another solid point! You're right, the test could have gotten flaky. I was under the false impression that gifs were considered animations and that cypress would [waitForAnimations](https://docs.cypress.io/guides/references/configuration#Actionability) to finish animating before executing commands.

Learned the hard way that gifs aren't animations (they don't animate DOM elements) and there's no easy way in JS to check if a gif has stopped.

What I've done now is:
- programmed the GIF to stop moving in 3200 ms.
- use cypress to get the DOM element
- wait for 4000 ms (guarantees that snapshot will be taken once gif stops)
- take the snapshot

While I recognize that using `cy.wait(<arbitrary-period>)` is an [anti-patten](https://docs.cypress.io/api/commands/wait#Time), I've been unable to find another way to check if the gif has stopped moving. In an ideal world, when the gif stops, we would hope that some HTML element or CSS property changed, so we can wait for that property instead of a time period, but that's not the case.

With my current implementation, the snapshot taken will always be the same. i.e. at the end stage of the gif; avoiding flakiness.

Reply:
Thanks for updating. I think this is a valid solution. Unfortunately, 4 seconds of waiting is too long. It would slow down our CI tests. I would propose either getting it down to under 1000ms (using fewer frames) or we leave out this snapshot test for now. Maybe add a `frames` parameter to the `create_gif` function to configure the number of frames, and use only one frame for the snapshot gif?

Reply:
Thanks Lukas! I'll get it down under 1000ms or use only one frame for the snapshot. Give me a day or two to fix the tests, I'm currently OOO

Reply:
Sounds good :)

Reply:
@LukasMasuch I adopted your idea of including a `frames` parameter, and for the snapshot we create a gif with only one frame and take the snapshot. There is 0 movement in the gif, so no chance of flakiness. Also, removed the 4 second delay.

Reply:
The snapshot will always look like this 
![image](https://user-images.githubusercontent.com/20672874/194481723-7f1e3cf9-8f28-4da0-a443-a4282f03037b.png)